### ISSUE TITLE
Request: Improve documentation for `openWhenHidden` property

### ISSUE THREAD

**Issue by jchiare** [Azure/fetch-event-source#36]
The default for `openWhenHidden` is false. This means that the SSE will abort on the client side if the document is hidden, however the default SSE behaviour is that the SSE will continue even if the document is hidden.

Since this is different then normal SSE, I think there should be something in the readme that says something along the lines of "To continue SSE streaming when browser is hidden, set `openWhenHidden` flag to true". I'm happy to do so if this is approved.

Thank you!

**Comment by martinseanhunt** [comment]
Thanks for this, I was wondering where that behaviour was coming from! I agree that documenting this functionality would be very useful.

**Comment by codeXLinkX** [comment]
Up

**Comment by devenbhooshan** [comment]
I wasted an hour figuring out why this wasn't working as expected. I expected `openWhenHidden` to be set `true` by default.  Please add this in the documentation.

**Comment by tpfau** [comment]
Please make this the default. 
This has driven me crazy for more than a day, and especially with web-apps requesting LLM streaming responses, this can cost money, when people tab out and into the tab and every time the request is resent....

**Comment by crown-li** [comment]
Good advice.
This parameter can only be found by looking at the source code

**Comment by ashi-dog** [comment]
I agree

**Comment by backas36** [comment]
yes, I think it's better to set openWhenHidden be true in default.

**Comment by easychen** [comment]
Agree , I spent two weeks looking for this problem

**Comment by chidera-ugo** [comment]
on ios safari it seems this still doesn't work if the safari is in the background when the message is received, is there a fix for this

**Comment by acodercat** [comment]
agree!

**Comment by att1106** [comment]
is there a way to keep fetching in the background when switching tabs? openWhenHidden: true keeps the connection alive but its not contitue fetching?

**Comment by wahsggx** [comment]
i agree！

**Comment by 100gle** [comment]
I met this issue...I think this property should be set `true` by default or point out in documentation. Otherwise it is like a bug 🤦‍♂️

**Comment by IAkumaI** [comment]
Oh my god, I thought I was going insane, looking for some chrome settings, digging into express configs and there is just a boolean setting for this.

**Comment by wangzhigang1999** [comment]
"Are you developers living in the Stone Age???

Seriously, you expect users to dig through the source code just to find out about the openWhenHidden parameter? And then you set the default to false—completely counterintuitive? Do you have any idea how many people got screwed over by this anti-user design and rage-quit debugging?

@Maintainers Do you think developers' time is worthless? Just in this issue alone, people wasted two weeks, lost money on retries, and scoured the internet for voodoo fixes—all because of a single undocumented boolean???

And the funniest part? This has been dragging on since March 2023, with everyone screaming for a fix, and all you do is reply with "Good advice" and ghost the issue? Even iOS compatibility problems are popping up, and you're still playing ostrich?

Might as well rename the repo to how-to-piss-off-developers-101 because your docs are about as clear as Morse code. Do you expect users to be mind readers or something? 💢"

**Comment by MarkusFra** [comment]
Just wanted to add some hours wasted to the counter in this issue. I agree, that the default should be `true`. At least it should be documented.
To be fair. It even says in the README

> In addition, this library also plugs into the browser's [Page Visibility API](https://developer.mozilla.org/en-US/docs/Web/API Page_Visibility_API) so the connection closes if the document is hidden (e.g., the user minimizes the window), and automatically retries with the last event ID when it becomes visible again. This reduces the load on your server by not having open connections unnecessarily (but you can opt out of this behavior if you want.)

But the option `openWhenHidden` itself is not mentioned.
